src/components/Garden/Metrics.js

```diff
@@ -128,37 +155,134 @@ function TokenBalance({ label, token, value, currency }) {
   )
 }
 
-function TokenPrice({ currency, token }) {
+function TokenPrice({ currency, oracleMode, token }) {
   const theme = useTheme()
-  const price = useHoneyswapTokenPrice(token.id)
+  const [price, setPrice] = useState(0)
+
+  const { config } = useGardenState()
+  const { stableToken } = config.conviction
+
+  const honeyswapPrice = useHoneyswapTokenPrice(token.id)
+
+  const [convertedAmount, loading, canUpdate] = useRequestAmount(
+    true,
+    1,
+    stableToken.id,
+    token.id
+  )
+
+  const [updatePriceOracle] = useUpdatePriceOracle(token, stableToken)
+
+  useEffect(() => {
+    if (loading) {
+      return
+    }
+
+    setPrice(oracleMode ? convertedAmount.valueOf() : honeyswapPrice)
+  }, [oracleMode, convertedAmount, honeyswapPrice, price, loading])
```

Comment:
In this case since we are doing only a calculation, we can memoize the value with a `useMemo` and we can remove the `price` state variable

```
  const price = useMemo(() => {
    if (loading) {
      return 0
    }

    return oracleMode ? convertedAmount.valueOf() : honeyswapPrice
  }, [convertedAmount, honeyswapPrice, loading, oracleMode])
```